worlds/overcooked2/Overcooked2Levels.py

```diff
@@ -0,0 +1,338 @@
+        assert False
+
+    # inclusive
+    def start_level_id(self) -> int:
+        if self == Overcooked2Dlc.STORY:
+            return 1
+        return 0
+
+    # exclusive
+    def end_level_id(self) -> int:
+        id = None
+        if self == Overcooked2Dlc.STORY:
+            id = 6*6 + 8  # world_count*level_count + kevin count
+        if self == Overcooked2Dlc.SURF_N_TURF:
+            id = 3*4 + 1
+        if self == Overcooked2Dlc.CAMPFIRE_COOK_OFF:
+            id = 3*4 + 3
+        if self == Overcooked2Dlc.NIGHT_OF_THE_HANGRY_HORDE:
+            id = 3*3 + 3 + 8
+        if self == Overcooked2Dlc.CARNIVAL_OF_CHAOS:
+            id = 3*4 + 3
+        if self == Overcooked2Dlc.SEASONAL:
+            id = 31
+
+        return self.start_level_id() + id
+
+    # Tutorial + Horde Levels + Endgame
+    def excluded_levels(self) -> List[int]:
+        if self == Overcooked2Dlc.STORY:
+            return [0, 36]
+
+        return []
+
+    def horde_levels(self) -> List[int]:
+        if self == Overcooked2Dlc.NIGHT_OF_THE_HANGRY_HORDE:
+            return [12, 13, 14, 15, 16, 17, 18, 19]
+        if self == Overcooked2Dlc.SEASONAL:
+            return [13, 15]
+
+        return []
+
+    def prep_levels(self) -> List[int]:
+        if self == Overcooked2Dlc.STORY:
+            return [1, 2, 5, 10, 12, 13, 28, 31]
+        if self == Overcooked2Dlc.SURF_N_TURF:
+            return [0, 4]
+        if self == Overcooked2Dlc.CAMPFIRE_COOK_OFF:
+            return [0, 2, 4, 9]
+        if self == Overcooked2Dlc.NIGHT_OF_THE_HANGRY_HORDE:
+            return [0, 1, 4]
+        if self == Overcooked2Dlc.CARNIVAL_OF_CHAOS:
+            return [0, 1, 3, 4, 5]
+        if self == Overcooked2Dlc.SEASONAL:
+            return [0, 1, 5, 6, 12, 14, 16, 17, 18, 22, 23, 24, 29]
+
+        return []
+
+
+class Overcooked2GameWorld(Enum):
+    ONE = 1
+    TWO = 2
+    THREE = 3
+    FOUR = 4
+    FIVE = 5
+    SIX = 6
+    KEVIN = 7
+
+    def as_str(self) -> str:
+        if self == Overcooked2GameWorld.KEVIN:
+            return "Kevin"
+
+        return str(int(self.value))
+
+    def get_sublevel_count(self) -> int:
+        if self == Overcooked2GameWorld.KEVIN:
+            return 8
+
+        return 6
+
+    def get_base_id(self) -> int:
+        if self == Overcooked2GameWorld.ONE:
+            return 1
+
+        prev = Overcooked2GameWorld(self.value - 1)
+        return prev.get_base_id() + prev.get_sublevel_count()
+
+    def name(self) -> str:
+        if self == Overcooked2GameWorld.KEVIN:
+            return "Kevin"
+
+        return "World " + self.as_str()
+
+
+
+class Overcooked2GenericLevel():
+    dlc: Overcooked2Dlc
+    level_id: int
+
+    def __init__(self, level_id: int, dlc: Overcooked2Dlc = Overcooked2Dlc("Story")):
+        self.dlc = dlc
+        self.level_id = level_id
+
+    def __str__(self) -> str:
+        return f"{self.dlc.value}|{self.level_id}"
+
+    def __repr__(self) -> str:
+        return self.__str__()
```

Comment:
slightly more pythonic to just do `return f"{self}"` as it's against PEP to call the special functions directly